pkg/attestlib/pkix_key.go

```diff
@@ -53,14 +53,18 @@ func generatePkixPublicKeyId(privateKey interface{}) (string, error) {
 		base64Dgst := base64.RawURLEncoding.EncodeToString(dgst[:])
 		return fmt.Sprintf("ni:///sha-256;%s", base64Dgst), nil
 	case *ecdsa.PrivateKey:
-		ecKey := privateKey.(*ecdsa.PrivateKey)
+		ecKey := key.(*ecdsa.PrivateKey)
 		publicKeyMaterial, err := x509.MarshalPKIXPublicKey(&ecKey.PublicKey)
 		if err != nil {
 			return "", errors.Wrap(err, "marshal ecdsa public key error")
 		}
 		dgst := sha256.Sum256(publicKeyMaterial)
 		base64Dgst := base64.RawURLEncoding.EncodeToString(dgst[:])
 		return fmt.Sprintf("ni:///sha-256;%s", base64Dgst), nil
+	case []byte:
+		dgst := sha256.Sum256(key.([]byte))
```

Comment:
this is for public key material?

Reply:
yes. I fixed it so it takes a hash of the bytes which is how it works in the private key cases.